x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleService.java

```diff
@@ -53,23 +53,20 @@
     private final PolicyStepsRegistry policyRegistry;
     private final IndexLifecycleRunner lifecycleRunner;
     private final Settings settings;
-    private final ThreadPool threadPool;
-    private Client client;
     private ClusterService clusterService;
     private LongSupplier nowSupplier;
     private SchedulerEngine.Job scheduledJob;
+    private boolean closed = false;
```

Comment:
I think the signal of whether the scheduler is shutdown should be 
contained within the SchedulerEngine. is this not the same as checking `engine.isShutdown()`:

```diff
--- a/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/scheduler/SchedulerEngine.java
+++ b/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/scheduler/SchedulerEngine.java
@@ -123,6 +123,11 @@ public class SchedulerEngine {
         jobs.forEach(this::add);
     }
 
+
+    public boolean isShutdown() {
+        return scheduler.isShutdown();
+    }
+
     public void stop() {
         scheduler.shutdownNow();
         try {
```